File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    
    const m = matrix.length;
    const n = matrix[0].length;
    const value =  (Math.pow(2, 31) -1) + 1
    
    for(let i = 0; i < m; i++) {
        for(let j = 0; j < n; j++) {
            if(matrix[i][j] == 0) {
                let c = 0;
                while(c < n) {
                    if(matrix[i][c] != 0) {
                        matrix[i][c] = value;
                    }
                    c++;
                }
                let r = 0;
                while( r < m) {
                    if(matrix[r][j] != 0) {
                        matrix[r][j] = value;
                    }
                    r++;
                }
            }
        }
    }
    //console.log(matrix)
    for(let i = 0; i < m; i++) {
        for(let j = 0; j < n; j++) {
            if(matrix[i][j] == value) matrix[i][j] = 0;
        }
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        final int m = matrix.length;
        final int n = matrix[0].length;

        boolean firstRow =  false;
        boolean firstCol = false;
        
        for(int  i = 0 ; i < m ;i++) {
            if(matrix[i][0] == 0){ firstCol = true; break;}
        }
        
        for(int j = 0 ; j < n; j++) {
            if(matrix[0][j] == 0) {
                firstRow = true;
                break;
            }
        }
        
        
        for(int i = 1; i < m; i++) {
            for(int j = 1; j < n; j++) {
                if(matrix[i][j] != 0) continue;
                matrix[0][j] = 0;
                matrix[i][0] = 0;
            }
        }
        for(int i = 1; i < m; i++) {
            for(int j = 1; j < n; j++) {
                if(matrix[i][0] == 0 || matrix[0][j] == 0) {
                    matrix[i][j] = 0;
                }
            }
        }
        if(firstRow) {
            int c = 0;
            while(c < n) {
                matrix[0][c] = 0;
                c++;
            }
        }
        if(firstCol) {
            int r = 0;
            while(r < m) {
                matrix[r][0] = 0;
                r++;
            }
        }
        

    }
}